documentation/cubic-pos-article.md

```diff
@@ -0,0 +1,156 @@
+
+When a validator misbehavior is detected by the protocol, the misbehaving validator is immediately jailed (unable to participate in consensus or make PoS transactions). The slash for the infraction is queued up to be processed at the epoch `n + unbonding_len` for an infraction committed in epoch `n`, allowing the protocol a sufficiently long time period (~21 days) to detect infractions. The cubic slashing algorithm is applied when processing the slash at the epoch `n + unbonding_len` and works as follows:
+
+1. For each slash queued to be processed in epoch `n`, collect the set of known slashes to be processed in epochs \[n-1, n+1\].
+2. Sum the fractional voting powers of the validator associated to each slash in this window:
+
+$$\text{sum}~ = \sum_{i \in \text{infractions}} \frac{\text{vp}_{i}}{\text{vp}_{tot}}$$
+
+Note that the voting power in the above expression is the voting power of the validator *when the infraction was committed*.
+
+3. The slash rate is then a function of the square of this sum, bounded below by some configurable minimum slash rate $r_{\text{min}}$ (with the corresponding plot):
+
+$$\max (~r_{\text{min}}~,~9*\text{sum}^2 ~).$$
+
+The factor of $9$ is included such that the slash rate maxes out at $100\%$ for total voting power of $1/3$. This is a critical point in the BFT consensus model, as a total colluding voting power of greater than $1/3$ can prevent the chain from achieving consensus. The cubic slashing rate is also plotted here for visual convenience:
+
+[<img src="cubic_slash.png" width="500"/>](cubic_slash.png)
+
+Once the slash rate is determined, the voting power of the jailed misbehaving validator can be immediately slashed and updated. This validator can then submit a transaction to become unjailed, which takes effect at the pipeline offset relative to the epoch of submission. Once the validator has been reinstated into the appropriate validator set depending on its new voting power, it and its delegators can resume bonding, unbonding, and redelegating.
+
+Namada's slashing system also guarantees that only delegators whose stake were used in a misbehaving validator's voting power during the epoch of infraction are slashed. If a delegator that contributed to a misbehaving validator's voting power at the time of infraction has since redelegated to a new validator by the time the infraction is discovered, then the protocol will still properly slash the delegator. This is an improved PoS guarantee offered by Namada, whereas in other chains using the Cosmos SDK's PoS module, there is no guarantee that all delegators contributing voting power to an infraction (and only these delegators) will be properly slashed for the infraction.
```

Comment:
```suggestion
Namada's slashing system also guarantees that only delegators whose stake were used in a misbehaving validator's voting power during the epoch of infraction are slashed. If a delegator that contributed to a misbehaving validator's voting power at the time of infraction has since redelegated to a new validator by the time the infraction is discovered, then the protocol will still properly slash the delegator. This is an improved PoS guarantee offered by Namada, whereas in other chains using the Cosmos SDK's PoS module, there is no guarantee that all delegators contributing voting power to an infraction (and only these delegators) will be properly slashed for the infraction. For further context, see [this issue](https://github.com/cosmos/cosmos-sdk/issues/1440).
```